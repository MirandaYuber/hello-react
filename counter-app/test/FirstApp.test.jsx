const { render } = require("@testing-library/react")
const { FirstApp } = require("../src/FirstApp")

describe('Pruebas en <FirstApp />', () => {
    test('Debe de hacer match con el snapshot', () => {
        const title = 'Hola soy Yuber'
        // renderiza el componente en memoria
        // El render, dentro de sus muchas fucnionalidades, expone un objeto con ciertas propiedades
        // una de ellas es container, que se parece a un nodo de HTML
        const {container, getByText} =  render(<FirstApp />)

        // Al tener un nodo HTML, podemos hacer la prueba creando una fotografia a ese nodo
        // Cuando ejecutemos esto la primera vez, guardar la captura en la carpeta __snapshots__ y sobre esa imagen 
        // se haran las comparaciones
        expect(container).toMatchSnapshot()
        
    })

    test('Debe de mostrar el titulo en un h1', () => {
        const title = 'Hola soy Yuber'

        // Dentro del render tenemos varios metodos para extraer diferentes cosas
        // getByText nos permite buscar un texto dentro del nodo HTML
        const {container, getByText} =  render(<FirstApp subTitle={title}/>)
        const h2 = container.querySelector('h2')

        expect(getByText(title)).toBeTruthy()
        expect(h2.innerHTML).toContain(title)
    })
    
})