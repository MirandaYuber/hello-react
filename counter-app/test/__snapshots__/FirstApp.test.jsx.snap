// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola Yuber Miranda
  </h1>
  <h2 />
  <p>
    JS
  </p>
  <code>
    {"name":"Yuber","age":23}
  </code>
</div>
`;
